src/thermo/src/utils/4C_thermo_discretization_runtime_output_params.hpp

```diff
@@ -0,0 +1,107 @@
+// This file is part of 4C multiphysics licensed under the
+// GNU Lesser General Public License v3.0 or later.
+//
+// See the LICENSE.md file in the top-level for license information.
+//
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#ifndef FOUR_C_THERMO_DISCRETIZATION_RUNTIME_OUTPUT_PARAMS_HPP
+#define FOUR_C_THERMO_DISCRETIZATION_RUNTIME_OUTPUT_PARAMS_HPP
+
+#include "4C_config.hpp"
+
+#include "4C_utils_parameter_list.fwd.hpp"
+
+FOUR_C_NAMESPACE_OPEN
+
+namespace Discret
+{
+  namespace Elements
+  {
+    /** \brief Input data container for output at runtime for thermo*/
+    class ThermoRuntimeOutputParams
```

Comment:
You can reduce all this boilerplate down to a simple struct 

```
struct ThermoRuntimeOutputParams
{
        /// whether to write pressure output
      bool output_temperature_state = false;

      /// whether to write the owner of elements
      bool output_element_owner = false;

      /// whether to write the element GIDs
      bool output_element_gid = false;

      /// whether to write the node GIDs
      bool output_node_gid = false;
};
```
which you fill with the correct values instead of calling `init` and `setup`.

You could also move the struct to the thermo_timint header (maybe even into the TimInt class), if you only need it there.